ravenframework/CodeInterfaceClasses/Abce/AbceCodeInterface.py

```diff
@@ -0,0 +1,169 @@
+      self._outputDirectory = os.path.join(os.path.dirname(settingsFile.getAbsFile()),'outputs',scenarioName)
+      #print("_outputDirectory", self._outputDirectory, "settingsFile", settingsFile.getAbsFile(), "scenarioName", scenarioName)
+      return None
+
+    #prepend
+    fullCommand = ''
+    fullCommand += clargs['pre']+' '
+    fullCommand += executable
+    index=None
+    #setup input files and output directory
+    self._outputDirectory = None
+    #inputs
+    for flag,elems in clargs['input'].items():
+      if flag == 'noarg':
+        continue
+      fullCommand += ' '+flag
+      for elem in elems:
+        ext, delimiter = elem[0], elem[1]
+        idx,fname = findSettingIndex(inputFiles,ext.strip('.'))
+        setOutputDir(fname)
+        fullCommand += delimiter + fname.getFilename()
+        if index == None:
+          index = idx
+    self.caseName = inputFiles[index].getBase()
+    outFile = 'out~'+self.caseName
+    if 'output' in clargs:
+      fullCommand+=' '+clargs['output']+' '+outFile
+    fullCommand+=' '+clargs['text']
+    fullCommand+=' '+clargs['post']
+    returnCommand = [('parallel',fullCommand)],outFile
+    print('Execution Command: '+str(returnCommand[0]))
+    return returnCommand
+
+  def createNewInput(self,currentInputFiles,origInputFiles,samplerType,**Kwargs):
+    """
+      This method is used to generate an input based on the information passed in.
+      @ In, currentInputFiles, list,  list of current input files (input files from last this method call)
+      @ In, oriInputFiles, list, list of the original input files
+      @ In, samplerType, string, Sampler type (e.g. MonteCarlo, Adaptive, etc. see manual Samplers section)
+      @ In, Kwargs, dictionary, kwarded dictionary of parameters. In this dictionary there is another dictionary called "SampledVars"
+            where RAVEN stores the variables that got sampled (e.g. Kwargs['SampledVars'] => {'var1':10,'var2':40})
+      @ Out, newInputFiles, list, list of newer input files, list of the new input files (modified and not)
+    """
+    infiles=[]
+    origfiles=[]
+    for inputFile in currentInputFiles:
+      if inputFile.getExt() in self.getInputExtension():
+        infiles.append(inputFile)
+    for inputFile in origInputFiles:
+      if inputFile.getExt() in self.getInputExtension():
+        origfiles.append(inputFile)
+    parser = GenericParser.GenericParser(infiles)
+    parser.modifyInternalDictionary(**Kwargs)
+    parser.writeNewInput(infiles,origfiles)
+    return currentInputFiles
+
+  def finalizeCodeOutput(self, command, codeLogFile, subDirectory):
+    """
+      Convert SQLite information to RAVEN's prefered formats [Pandas DataFrame]
+      Joins together two different SQLite files and also reorders it a bit.           @ In, command, ignored
+      @ In, codeLogFile, ignored
+      @ In, subDirectory, string, the subdirectory where the information is.
+      @ Out, directory, string, the assets results
```

Comment:
it looks like the output is a dictionary, maybe?

Reply:
Fixed.